ui/src/app/utils/runtime-hooks.spec.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+  Disk,
+  DisksApi,
+  DiskType,
+  GpuConfig,
+  RuntimeApi,
+  RuntimeConfigurationType,
+  RuntimeStatus,
+} from 'generated/fetch';
+
+import { DisksApiStub } from '../../testing/stubs/disks-api-stub';
+import { waitFor } from '@testing-library/react';
+// Hooks cannot be tested outside of a functional component, so we need to use renderHook from @testing-library/react-hooks
+// This is explained further here: https://github.com/testing-library/react-hooks-testing-library?tab=readme-ov-file#the-problem
+// TODO: Once we update to React 18 and a more recent RTL version,
+//  these imports can be replaced with the standard RTL and
+// @testing-library/react-hooks can be uninstalled
+import { act, renderHook } from '@testing-library/react-hooks';
+import {
+  registerApiClient,
+  runtimeApi,
+} from 'app/services/swagger-fetch-clients';
+
+import {
+  defaultDataProcRuntime,
+  defaultGceRuntime,
+  defaultGceRuntimeWithPd,
+  defaultRuntime,
+  RuntimeApiStub,
+} from 'testing/stubs/runtime-api-stub';
+import { workspaceDataStub } from 'testing/stubs/workspaces';
+
+import { LeoRuntimeInitializer } from './leo-runtime-initializer';
+import { DATAPROC_MIN_DISK_SIZE_GB } from './machines';
+import { useCustomRuntime } from './runtime-hooks';
+import { runtimeDiskStore, runtimeStore } from './stores';
+
+describe(useCustomRuntime.name, () => {
+  let disksApiStub;
+  let runtimeApiStub;
+  let deleteRuntimeSpy;
+  let initializerSpy;
+  let updateRuntimeSpy;
+  let currentRuntime;
+  let deleteDiskSpy;
+
+  beforeEach(() => {
+    // jest.useFakeTimers();
+    // useCustomRuntime requires a runtime to be loaded in the
+    // runtimeStore. All tests start with the default runtime.
+    currentRuntime = defaultRuntime();
+    runtimeStore.set({
+      workspaceNamespace: workspaceDataStub.namespace,
+      runtime: currentRuntime,
+      runtimeLoaded: true,
+    });
+
+    runtimeApiStub = new RuntimeApiStub();
+    registerApiClient(RuntimeApi, runtimeApiStub);
+    deleteRuntimeSpy = jest.spyOn(runtimeApi(), 'deleteRuntime');
+    initializerSpy = jest.spyOn(LeoRuntimeInitializer, 'initialize');
+    updateRuntimeSpy = jest.spyOn(runtimeApi(), 'updateRuntime');
+    disksApiStub = new DisksApiStub();
+    registerApiClient(DisksApi, disksApiStub);
+    deleteDiskSpy = jest.spyOn(disksApiStub, 'deleteDisk');
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const testUseCustomRuntime = () => {
+    const { result } = renderHook(() =>
+      useCustomRuntime('workspaceNamespace', null)
+    );
+
+    // Result's current property is the return value of the hook
+    return result.current;
+  };
+
+  const setCurrentDisk = (d: Disk) => {
```

Comment:
This is also in runtime-configuration-panel. Could it be moved to a common location?